src/compiler/backend/riscv64/code-generator-riscv64.cc

```diff
@@ -2691,9 +2691,16 @@ void CodeGenerator::AssembleSwap(InstructionOperand* source,
       } else {
         DCHECK(destination->IsFPStackSlot());
         MemOperand dst = g.ToMemOperand(destination);
-        __ Move(temp, src);
-        __ LoadDouble(src, dst);
-        __ StoreDouble(temp, dst);
+        if (rep == MachineRepresentation::kFloat32) {
+          __ fmv_s(temp, src);
```

Comment:
I'd prefer to use a higher level interface here and try to only call `Assembler` APIs from the `TurboAssembler`. I'd suggest either adding a third parameter to the floating point `Move` or creating a `MoveF` and a `MoveD`. 

@penguinwu, I think you did some refactoring of these moves before. Your thoughts?

Reply:
Similar to `LoadDouble/StoreDouble`, we can add `MoveDouble` and `MoveFloat`.